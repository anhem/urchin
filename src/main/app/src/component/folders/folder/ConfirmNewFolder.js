import React, {Component} from 'react'
import FieldError from "../../FieldError";

class ConfirmNewFolder extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setEncryptedFolderPassphrase(data);
    };

    confirm = () => {
        let folderId = this.props.createdFolder.id;
        this.props.callbacks.confirmEncryptedFolder(folderId, this.props.confirmNewFolder.passphrase);
    };

    render() {
        let passphrase = this.props.confirmNewFolder.passphrase || '';
        let passphraseToMatch = this.props.createdFolder !== undefined ? this.props.createdFolder.passphrase : undefined;

        let content = null;
        if (passphraseToMatch) {
            content = <div>
                <div>
                    Your folder has been created.
                    This is the randomly generated passphrase for your folder:
                </div>
                <p data-view="passphraseToMatch">
                    {passphraseToMatch}
                </p>
                <div>
                    Make sure to write down this passphrase and store it in a safe place!
                </div>
                <input name="passphrase"
                       type="password"
                       value={passphrase}
                       onChange={this.update}/>
                <FieldError fieldErrors={this.props.fieldErrors}
                            field="passphrase"/>
                <button data-view="confirm"
                        className="confirm-new-folder__confirm-btn"
                        onClick={this.confirm}>
                    Confirm
                </button>
            </div>;
        } else {
            content = <div>
                Undesired state! Passphrase is missing. Please delete newly created folder and try again.
            </div>
        }

        return (
            <div data-view="confirmNewFolder"
                 className="confirm-new-folder">
                {content}
            </div>
        )
    }
}

export default ConfirmNewFolder