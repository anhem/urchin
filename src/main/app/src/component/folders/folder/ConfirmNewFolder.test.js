import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import ConfirmNewFolder from "./ConfirmNewFolder";

describe('ConfirmNewFolder', () => {

    let callbacks = {
        setEncryptedFolderPassphrase: jest.fn(),
        confirmEncryptedFolder: jest.fn(),
    };

    let props = {
        confirmNewFolder: {
            passphrase: 'random'
        },
        createdFolder: {
            id: 1,
            passphrase: 'randomString',
        },
        callbacks: callbacks
    };

    beforeEach(() => {
        Object.entries(callbacks).forEach(([name, fn]) => {
            fn.mockClear();
        });
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<ConfirmNewFolder {...props}/>))).toMatchSnapshot();
    });

    it('setEncryptedFolderPassphrase is called when passphrase changes', () => {
        let component = shallow(<ConfirmNewFolder {...props}/>);
        component.find('[name="passphrase"]').simulate('change', {target: {value: 'a'}});

        expect(callbacks.setEncryptedFolderPassphrase).toHaveBeenCalledTimes(1);
    });

    it('confirmEncryptedFolder is called when clicking confirm', () => {
        let component = shallow(<ConfirmNewFolder {...props}/>);
        component.find('[data-view="confirm"]').simulate('click');

        expect(callbacks.confirmEncryptedFolder).toHaveBeenCalledWith(props.createdFolder.id, props.confirmNewFolder.passphrase);
    });
});
