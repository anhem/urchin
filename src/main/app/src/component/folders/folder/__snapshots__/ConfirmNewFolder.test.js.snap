// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmNewFolder match snapshot 1`] = `
<div
  className="confirm-new-folder"
  data-view="confirmNewFolder"
>
  <div>
    <div>
      Your folder has been created. This is the randomly generated passphrase for your folder:
    </div>
    <p
      data-view="passphraseToMatch"
    >
      randomString
    </p>
    <div>
      Make sure to write down this passphrase and store it in a safe place!
    </div>
    <input
      name="passphrase"
      onChange={[Function]}
      type="password"
      value="random"
    />
    <FieldError
      field="passphrase"
    />
    <button
      className="confirm-new-folder__confirm-btn"
      data-view="confirm"
      onClick={[Function]}
    >
      Confirm
    </button>
  </div>
</div>
`;
